crates/sp-domains/src/lib.rs

```diff
@@ -170,6 +171,12 @@ impl DomainId {
     }
 }
 
+impl Display for DomainId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        write!(f, "domain-{}", self.0)
```

Comment:
This doesn't seem right. It is a number, so should be printed as a number. Printing it as `domain-{}` is non-idiomatic IMO.

Reply:
`DomainId` is a wrapper of number, we will still need `format!("domain-{}", domain_id)` if not add `domain-` here, anyway both ways lgtm

Reply:
Where do we need it?

Reply:
https://github.com/subspace/subspace/pull/2161/files#diff-1accbe4a6fdef4ef2d894523cce59e63eb22e2578637e2458e4c10df59f6fa17R585

Reply:
Well, as I commented it should be `DomainId` there, not `String`, then it is not needed. Even if you do want a string, there is no need to add `domain-` prefix as long as it is consistent.